rootfs-builder/README.md

```diff
@@ -0,0 +1,62 @@
+# Building a rootfs for Kata Containers Guest OS #
+
+The Kata Containers guest OS is created using `rootfs.sh`.
+
+## Supported base OS ##
+
+The `rootfs.sh` script support different based OS `rootfs`. To build
+a `rootfs` based on a supported OS run:
+
+```
+$./rootfs.sh <distro>
+```
+
+To check the supported `rootfs` based OS run `$rootfs-builder/rootfs.sh
+-h`, it will show the supported values of `<distro>`
+
+
+## Adding support for new base OS ##
+
+The script `rootfs.sh` will it check for immediate sub-directories
+containing the following expected files structure:
+
+- A `bash(1)` script called `rootfs_lib.sh`
+
+  This file must contain a function called `build_rootfs()` this function
+  must receive as first argument the path where the `rootfs` will be
+  populated. Path: `rootfs-builder/<distro>/rootfs_lib.sh`.
+
+
+- A `bash(1)` file `config.sh`
+
+  This represents the specific configuration for `<distro>`. It must
+  provide configuration specific variables for user to modify as needed.
+  The `config.sh` file will be loaded before executing `build_rootfs()` to
+  provide all the needed configuration to the function. Path:
+  `rootfs-builder/<distro>/config.sh`.
+
+To create a directory with the expected file structure run:
+
+```
+make -f template/Makefile  ROOTFS_BASE_NAME=my_new_awesome_rootfs
+```
+
+After run the command above, a new directory will be created in
+`rootfs-builder/my_new_awesome_rootfs/`. To verify it is one of the
+options to build a `rootfs` run `./rootfs.sh -h`, it will show
+`my_new_awesome` as one of the options to use it for:
+
+```
+./rootfs.sh <distro>
+```
+
+Now that a new directory structure was created is need to:
+
+- If needed , add configuration variables to `rootfs-builder/my_new_awesome_rootfs/config.sh`
+- Implement the stub `build_rootfs()` function from `rootfs-builder/my_new_awesome_rootfs/rootfs_lib.sh`
+
+### Expected `rootfs` directory content ###
+
+After the function `build_rootfs` is called the script expects the
+`rootfs` directory contain a `init` binary in `/sbin/init` and
```

Comment:
s/directory contain a/directory to contain `/sbin/init` and `/sbin/kata-agent` binaries./